parcels/grid.py

```diff
@@ -78,7 +79,16 @@ def from_netcdf(cls, filenames, variables, dimensions, **kwargs):
             fields[var] = Field.from_netcdf(var, dimensions, dsets, **kwargs)
         u = fields.pop('U')
         v = fields.pop('V')
-        return cls(u, v, u.depth, u.time, fields=fields)
+
+        # assign time_origin if the time dimension has units and calendar
+        try:
+            time_units = dsets[0][dimensions['time']].units
+            calendar = dsets[0][dimensions['time']].calendar
+            time_origin = num2date(0, time_units, calendar)
```

Comment:
OK, so  as I understand it, the whole idea of the `num2date` function of netCDF4 is to convert everything in the same units (seconds in this case). 

I just pushed a commit where `time_origin` is in the `Field` class. So your situation, where one field is in seconds and the other in days, shouldn't happen anymore with this new PR. Could you check, @Jacketless ?